dts/bindings/psi5/psi5-controller.yaml

```diff
@@ -0,0 +1,153 @@
+# Copyright 2025 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: Peripheral Sensor Interface (PSI5) Controller
+
+include: base.yaml
+
+properties:
+  reg:
+    required: true
+
+child-binding:
+  description: |
+    Each child node defines the configuration of a Peripheral Sensor Interface (PSI5) channel
+
+  properties:
+    reg:
+      type: int
+      required: true
+      description: Channel identifier.
+
+    async-mode:
+      type: boolean
+      description: |
+        Determines the channel operation mode. When set to true, the channel operates in
+        asynchronous mode with only the receive function active. When set to false,
+        the channel operates in synchronous mode with both transmit and receive functions active.
+
+    period-sync-pulse-us:
+      type: int
+      required: true
+      description: |
+        Specifies the period of the internally generated synchronization pulses, measured in
+        microseconds (us). This value determines the length of each synchronization pulse used
+        in the system.
+
+    pulse-width-0-us:
+      required: true
+      type: int
+      description: |
+        Specifies the duration of the pulse width for a data bit value '0', measured in
+        microseconds (us).
+
+    pulse-width-1-us:
+      required: true
+      type: int
+      description: |
+        Specifies the duration of the pulse width for a data bit value '1', measured in
+        microseconds (us).
+
+    decoder-start-offset-us:
+      type: int
+      required: true
+      description: |
+        Specifies the duration for which the Manchester decoder remains inactive after the
+        falling edge of a synchronization pulse, measured in microseconds (us). This value
+        determines the delay before the decoder starts processing incoming signals again.
+
+    tx-mode:
+      type: string
+      required: true
+      enum:
+        - short-frame-31
+        - short-frame-5
+        - long-frame-31
+        - long-frame-5
+        - x-long-frame-31
+        - x-long-frame-5
+        - xx-long-frame
+        - non-standard-frame
+      description: |
+        Specifies the transmitter mode. Each mode defines the frame length and
+        the start condition for data transmission:
+          - short-frame-31: Short Frame (V1.3) with 31 "1s" as the start condition
+          - short-frame-5: Short Frame (V1.3) with 5 "0s" as the start condition
+          - long-frame-31: Long Frame (V1.3) with 31 "1s" as the start condition
+          - long-frame-5: Long Frame (V1.3) with 5 "0s" as the start condition
+          - x-long-fram-31: X-Long Frame (V1.3) with 31 "1s" as the start condition
+          - x-long-frame-5: X-Long Frame (V1.3) with 5 "0s" as the start condition
+          - xx-long-frame: XX-Long (V2.0)
+          - non-standard-frame: Non Standard Length
+
+    num-rx-buf:
```

Comment:
it seems to be this is vendor specific property?

Reply:
I moved this to vendor S32 binding